vault/plugin_reload.go

```diff
@@ -0,0 +1,103 @@
+package vault
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+// reloadPluginMounts reloads provided mounts, regardless of
+// plugin name, as long as the backend type is of plugin.
+func (c *Core) reloadMatchingPluginMounts(mounts []string) error {
+	c.mountsLock.Lock()
+	defer c.mountsLock.Unlock()
+
+	for _, mount := range mounts {
```

Comment:
This can partially reload some backends before failing. I wonder if this should be an all-or-nothing operation, where reload executed only after a check on all mounts successful.